spring-integration-core/src/test/java/org/springframework/integration/handler/MethodInvokingMessageProcessorTests.java

```diff
@@ -583,26 +586,30 @@ public void testPerformanceSpelVersusInvocable() throws Exception {
 		StopWatch stopWatch = new StopWatch("SpEL vs Invocable Performance");
 
 		stopWatch.start("SpEL");
-		for (int i = 0; i < 10000; i++) {
+		int count = 10_000;
+		for (int i = 0; i < count; i++) {
 			processor.processMessage(message);
 		}
 		stopWatch.stop();
 
 		processor = new MethodInvokingMessageProcessor(service, method);
 
 		stopWatch.start("Invocable");
-		for (int i = 0; i < 10000; i++) {
+		for (int i = 0; i < count; i++) {
 			processor.processMessage(message);
 		}
 		stopWatch.stop();
 
+		// Update the parser configuration compiler mode
+		SpelParserConfiguration config = TestUtils.getPropertyValue(processor,
+				"delegate.handlerMethod.EXPRESSION_PARSER.configuration", SpelParserConfiguration.class);
+		new DirectFieldAccessor(config).setPropertyValue("compilerMode", SpelCompilerMode.IMMEDIATE);
```

Comment:
Well, since according to your comment the SpEL of this test isn't compilable, we might should consider to switch to something really compilable.
When I use `messageOnly()` method I have this result:
```
00202  037%  SpEL
00187  034%  Invocable
00164  030%  Compiled SpEL
```
So, somehow for the whole `Message<>` SpEL is faster 😄 

When I use `integerMethod()` I get this:
```
00179  051%  SpEL
00068  020%  Invocable
00101  029%  Compiled SpEL
```

So, looks like there is some performance bottleneck when we really use  `Message` parameter...

The `System.clearProperty("spring.expression.compiler.mode");` must be removed from this test now. Redundant.